kadi/commands/states.py

```diff
@@ -1388,24 +1387,43 @@ class NormalSunTransition(ManeuverTransition):
     """
     Same as ``ManeuverTransition`` except perform maneuver to NSM.
 
-    This is based on a pure-pitch maneuver from current attitude).  It also
-    changes ``pcad_mode`` to NSUN.
+    This uses ska_sun.get_nsm_attitude() to get the NSM attitude, potentially at
+    an offset pitch like 160 degrees.  It also changes ``pcad_mode`` to NSUN.
     """
 
     command_attributes = {"tlmsid": "AONSMSAF"}
+    pcad_mode = "NSUN"
     state_keys = PCAD_STATE_KEYS
 
     @classmethod
-    def callback(cls, date, transitions, state, idx):
+    def set_transitions(cls, transitions, cmds, start, stop):
+        state_cmds = cls.get_state_changing_commands(cmds)
+
+        # Set the maneuver transition for each state-changing command. The AONSMSAF or
+        # ACPCSFSU commands have an optional pitch parameter corresponding to the Params
+        # value of the params for the NSM or Safe Mode command events. If not provided
+        # then use 90 degrees.
+
+        # The functools.partial is a little odd in that it passes the class directly.
+        # Ideally we would use functools.partial of a @classmethod but that doesn't seem
+        # to work.
+        for cmd in state_cmds:
+            pitch = cmd["params"].get("pitch", 90)
+            transitions[cmd["date"]]["maneuver_transition"] = functools.partial(
+                cls.callback, cls, pitch
+            )
+
+    @staticmethod
+    def callback(cls, pitch, date, transitions, state, idx):  # noqa: PLW0211
         """
         This is a transition function callback.
 
         It directly sets the state pcad_mode to NSUN and target quaternion to
```

Comment:
Does this docstring also need a tweak for appropriate PCAD mode instead of NSUN?

Reply:
Done.